### ISSUE TITLE
[BUG] FZF-TAB TMUX NOT WORKING

### ISSUE THREAD

**Issue by Taransaurus** [Aloxaf/fzf-tab#162]
**Describe the bug**
Cannot use tmux's popup, permissions error.

**To Reproduce**
1. Open tmux
2. Press <kbd>Tab</kbd>
3. See error

**Expected behavior**
A working fzf-tab popup in tmux.
**Screenshots**
![image](https://user-images.githubusercontent.com/57827456/99868898-52be8200-2bf9-11eb-8829-007dc8a27cf9.png)

**Environment:**
 - OS: Arch Linux
 - zsh version: 5.8
 - fzf-tab version: latest
 - tmux version: 3.2-rc3

**Comment by Aloxaf** [comment]
That's strange. You are the second one to report this issue.

I add a new widget `fzf-tab-debug`. Please update and try following steps, then upload the log file here.

```
> bindkey "^I" fzf-tab-debug # bind it to TAB
> : <TAB> # Press TAB here 
```

BTW, you shouldn't run `ftb-tmux-popup` directly, it's an internal function of fzf-tab.

**Comment by Taransaurus** [comment]
> That's strange. You are the second one to report this issue.
> 
> I add a new widget `fzf-tab-debug`. Please update and try following steps, then upload the log file here.
> 
> ```
> > bindkey "^I" fzf-tab-debug # bind it to TAB
> > : <TAB> # Press TAB here 
> ```
> 
> BTW, you shouldn't run `ftb-tmux-popup` directly, it's an internal function of fzf-tab.

It's just a test btw, also thanks for replying xd

**Comment by Taransaurus** [comment]
https://gist.github.com/b798f760fea822a7c96196c689dc5b58
Here's the output.

**Comment by Aloxaf** [comment]
Did you install fzf through zinit?

Please update and try again.

**Comment by Taransaurus** [comment]
> Did you install fzf through zinit?
> 
> Please update and try again.

Didn't work.

**Comment by Aloxaf** [comment]
> > Did you install fzf through zinit?
> > Please update and try again.
> 
> Didn't work.
> Also update what? fzf-tab?

Sorry, I mean I suspect that you install fzf through zinit, which will break `ftb-tmux-popup`.

I have fixed it in https://github.com/Aloxaf/fzf-tab/commit/58f95ac5c8c02c85286ba5a74adc057bbee7eb99, please update fzf-tab and try again.

**Comment by Taransaurus** [comment]
Still ;-; https://gist.github.com/Taransaurus/130b72133ca51b4a7f4c50aa64d7906e
I did `rm -rf` Aloxaf*fzf-tab in .zinit plugins folder and reinstall fzf-tab :(

**Comment by Aloxaf** [comment]
Ummm, strange. Please follow these steps:

1. paste the output of `tmux popup -R 'echo $SHELL'`
2. run `autoload +X ftb-tmux-popup && functions[ftb-tmux-popup]=${functions[ftb-tmux-popup]/command rm/:}`
3. trigger completion with `ls -<TAB>` or something else
4. paste the output of `ls -l /tmp/zsh-fzf-tab-taran`
5. if there is a file named `fzf-xxxx`, paste its content and run it with `zsh /tmp/zsh-fzf-tab-taran/fzf-$$` and check if fzf window will appear.
6. if fzf window appears, run `tmux popup -KE -R "zsh /tmp/zsh-fzf-tab-taran/fzf-$$"` and check if popup will appear.

**Comment by Taransaurus** [comment]
https://streamable.com/81usi3
There is

**Comment by Aloxaf** [comment]
It's empty? Really strange.

Can you run `autoload +X ftb-tmux-popup && functions[ftb-tmux-popup]=${functions[ftb-tmux-popup]/-KE/-K}` and then trigger completion? I think this time you will get a popup (with error message in it).

And is `/tmp/zsh-fzf-tab-taran/` writable? Can you create a file with `echo abc > /tmp/zsh-fzf-tab-taran/example`?

**Comment by Taransaurus** [comment]
https://streamable.com/tup7ea

**Comment by Aloxaf** [comment]
Can you provide your zshrc?

**Comment by Taransaurus** [comment]
> Can you provide your zshrc?

It's included in my dotfiles
![zsh](https://github.com/Taransaurus/bspwm-dotfiles/tree/master/zsh)

**Comment by nathanph** [comment]
Updating `ftb-tmux-popup` like so seems to work for me for some reason:

```
  tmux popup -x $popup_x -y $popup_y \
       -w $popup_width -h $popup_height \
       -K -R "zsh $tmp_dir/fzf-$$"
```

**Comment by Taransaurus** [comment]
> Updating `ftb-tmux-popup` like so seems to work for me for some reason:
> 
> ```shell
>   tmux popup -x $popup_x -y $popup_y \
>        -w $popup_width -h $popup_height \
>        -KE -R "zsh $tmp_dir/fzf-$$"
> ```
> 
> This is probably technically better:
> 
> ```shell
>   tmux popup -x $popup_x -y $popup_y \
>        -w $popup_width -h $popup_height \
>        -KE -R "${SHELL} $tmp_dir/fzf-$$"
> ```

**Comment by Taransaurus** [comment]
It worked for me either, thanks

**Comment by Taransaurus** [comment]
Bye i'm gonna close this, thanks for the help.

**Comment by Aloxaf** [comment]
Oh, I think I finally find the reason.

You guys must use `dash` or something else as the default `sh`.

**Comment by Aloxaf** [comment]
Fixed in https://github.com/Aloxaf/fzf-tab/commit/2f1ca128d92064ccb8984add8eeb39811c9e0b95.
Please update fzf-tab and try again.

**Comment by nathanph** [comment]
> Oh, I think I finally find the reason.
> 
> Your default `sh` may be `dash` or something else.

I'm running Ubuntu so that makes sense for me.

**Comment by fedoranvar** [comment]
By this date, this issue repeated with latest commit `0c36bdcf`.

I've reverted ftb-tmux-popup to this commit`2f1ca12` and all working again

**Comment by Aloxaf** [comment]
> By this date, this issue repeated with latest commit `0c36bdcf`.

What's the error message? Is it the same as this issue?

**Comment by TheAllSeeing** [comment]
I had a problem with this (running Ubuntu 20.04), that weirdly solved most of itself as I was trying to debug it

At first, it could not find ftb-tmux-popup. I have no idea why that happened. Pretty soon after it did just didn't recognize the popup command. I ran tmux 3.0, so I upgraded to the most recent version (3.2) and that solved _that_ error, resulting in unrecognized options -R and -K (removed; see [here](https://github.com/tmux/tmux/issues/1842#issuecomment-788955226).). I removed these options from ftb-tmux-popup and that seemed to solve the issue.
